examples/granular.rs

```diff
@@ -1,3 +1,5 @@
+use rand::rngs::ThreadRng;
```

Comment:
Where can I read more about your granular synthesis project? I'm intrigued

Reply:
That's quite a large topic I guess, but the general idea is quite simple at the end, just create some sound by concatenating small pieces of sound:
- https://en.wikipedia.org/wiki/Granular_synthesis
- http://erickuehnl.com/1995/12/30/granular-synthesis/

A very important aspect to achieve clean results is to have very accurate scheduling, that's why we need some scheduler to work with a logical time that is decoupled from the audio io callbacks and other timing functions. You can find a good explanation of the why and how there: https://web.dev/audio-scheduling/, the article is very JS oriented, but I think the main ideas holds for any language, i.e. you can assume you would have the same inaccuracy and drifting problems with `thread::sleep(...);` (I tested, it does :)

Reply:
If you want to play with an interactive example: https://apps.ismm.ircam.fr/demos/mini-catart (just click and move over the dots, the "Hendrix" file works quite well). It's a slightly different approach, called concatenative synthesis, where the consecutive grains are chosen according to some analysis results (rms and pitch/noisiness in the example)

The other examples I have are currently broken for some reason, but I could update them actually